cli/command/load_balancer_test.go

```diff
@@ -6,41 +6,323 @@ import (
 	"github.com/quintilesims/layer0/common/models"
 )
 
-func TestCreateLoadBalancer(t *testing.T) {
+func TestAddPortToLoadBalancer(t *testing.T) {
 	base, ctrl := newTestCommand(t)
 	defer ctrl.Finish()
 
-	ports := []models.Port{}
+	base.Resolver.EXPECT().
+		Resolve("load_balancer", "lname").
+		Return([]string{"lid"}, nil)
+
+	ports := []models.Port{
+		models.Port{
+			ContainerPort: 80,
+			HostPort:      80,
+			Protocol:      "tcp",
+		},
+	}
 	healthCheck := models.HealthCheck{}
 
+	base.Client.EXPECT().
+		ReadLoadBalancer("lid").
+		Return(&models.LoadBalancer{
+			HealthCheck:      healthCheck,
+			LoadBalancerID:   "lid",
+			LoadBalancerName: "lname",
+			Ports:            ports,
+		}, nil)
+
+	ports = append(ports, models.Port{
+		ContainerPort: 81,
+		HostPort:      81,
+		Protocol:      "tcp",
+	})
+
+	req := models.UpdateLoadBalancerRequest{
+		LoadBalancerID: "lid",
+		Ports:          &ports,
+		HealthCheck:    &healthCheck,
+	}
+
+	base.Client.EXPECT().
+		UpdateLoadBalancer(req).
+		Return("jid", nil)
+
+	base.Client.EXPECT().
+		ReadJob("jid").
+		Return(&models.Job{
+			Status: "Completed",
+			Result: "lid",
+		}, nil)
+
+	base.Client.EXPECT().
+		ReadLoadBalancer("lid").
+		Return(&models.LoadBalancer{}, nil)
+
+	loadBalancerCommand := NewLoadBalancerCommand(base.Command())
+	c := getCLIContext(t, args{"lname", "81:81/tcp"}, flags{})
+	if err := loadBalancerCommand.addport(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestAddPortToLoadBalancer_UserInputError(t *testing.T) {
+}
+
+func TestCreateLoadBalancer(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	base.Resolver.EXPECT().
+		Resolve("environment", "ename").
+		Return([]string{"eid"}, nil)
+
+	ports := []models.Port{
+		models.Port{
+			ContainerPort: 80,
+			HostPort:      80,
+			Protocol:      "tcp",
+		},
+	}
+	healthCheck := models.HealthCheck{}
```

Comment:
we should test a non zero-value `healthcheck` during `create`

Reply:
:white_check_mark:

Reply:
didn't realize this was in `TestAddPort`. In that case, I think having any sort of `healthcheck` in this test is a sort of red herring - should be removed IMO